### DISCUSSION TITLE
[Feature request] Notification channel type that can exec local script / command

### DISCUSSION THREAD

**Discussion by mrh666** [grafana/grafana#36980] [Alerting]
That will be really useful to support SNMP or at least exec on the local machine in alerting notifications. Is there is any chance you'll support it in near feature?

**Comment by bergquist** [comment]
We might add support for executing commands when an alert is firing. But im not sure about when.

No idea about SNMP for notifications. I guess I'll have to read the spec first :)

**Comment by rustyricky** [comment]
+1

**Comment by stromnet** [comment]
This would be a simple workaround for integrating with all sorts of systems which does not implement your "webhook" protocol, for example triggering nsca to submit to nagios, or op5 REST API which uses another data format. A custom body for webhooks would also be usable option in the later case, but script execution is definitely more versatile. (edit: as explained in #6956)

**Comment by mo8Zomo0** [comment]
Is there any progress? This sounds like a feature which should be one of the fist to integrate to me as @stromnet mentioned might be a suitable workaround for integrations currently not present.

**Comment by bergquist** [comment]
No progress from the Grafana team. Pull requests are more than welcome.

**Comment by Marctraider** [comment]
Anything? Looks like a basic feature missing

**Comment by siegmund42** [comment]
It's basically all about adding a new Notifier in `pkg/services/alerting/notifiers/`, right?

**Comment by 16g** [comment]
> It's basically all about adding a new Notifier in `pkg/services/alerting/notifiers/`, right?

Sounds like truth. Do you have any news?

**Comment by dpsenner** [comment]
Seconding this feature request. I would use this to fire custom actions on the server that need to be scripted to work properly. Implementing a rpc backend is way more effort in comparison.

**Comment by marzlarz** [comment]
Would also find this very useful

**Comment by dorchain** [comment]
I second the feature request.

**Comment by zottelchin** [comment]
Hi, I would really appreciate the option of an executable contact point. Are there any updates regarding this feature?

**Comment by natellium** [comment]
Hey there! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

I will be closing this discussion, but if you’re still interested in this request being implemented please don’t hesitate to open a [feature request](https://github.com/grafana/grafana/issues/new?assignees=&labels=&projects=&template=1-feature_requests.md). Sorry for the inconvenience, and thanks for your contributions!

cc @grafana/alerting-squad

**Comment by 16g** [comment]
nu pzdc:(
